app/src/main/java/io/homeassistant/companion/android/widgets/button/ButtonWidget.kt

```diff
@@ -141,25 +145,40 @@ class ButtonWidget : AppWidgetProvider() {
 
         super.onReceive(context, intent)
         when (action) {
+            CALL_SERVICE_AUTH -> authThenCallConfiguredService(context, appWidgetId)
             CALL_SERVICE -> callConfiguredService(context, appWidgetId)
             RECEIVE_DATA -> saveServiceCallConfiguration(context, intent.extras, appWidgetId)
             Intent.ACTION_SCREEN_ON -> updateAllWidgets(context)
         }
     }
 
+    private fun authThenCallConfiguredService(context: Context, appWidgetId: Int) {
+        Log.d(TAG, "Calling authentication, then configured service")
+
+        val intent = Intent(context, WidgetAuthenticationActivity::class.java)
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+        context.startActivity(intent)
+    }
+
     private fun getWidgetRemoteViews(context: Context, appWidgetId: Int): RemoteViews {
         // Every time AppWidgetManager.updateAppWidget(...) is called, the button listener
         // and label need to be re-assigned, or the next time the layout updates
         // (e.g home screen rotation) the widget will fall back on its default layout
         // without any click listener being applied
 
+        val widget = buttonWidgetDao.get(appWidgetId)
+        var auth = false
+
+        if (widget?.requireAuthentication == true) {
+            auth = true
+        }
```

Comment:
No need to change it later, so it's better to set it when initialized.

```suggestion
        val auth = widget?.requireAuthentication == true
```

Reply:
OK